cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/preview/PreviewHttpHandler.java

```diff
@@ -0,0 +1,241 @@
+
+    try {
+      Files.createDirectories(previewDirPath);
+      File previewDir = Files.createTempDirectory(previewDirPath, applicationId.getApplication()).toFile();
+      previewcConf.set(Constants.CFG_LOCAL_DATA_DIR, previewDir.getAbsolutePath());
+      previewcConf.set(Constants.Dataset.DATA_DIR, previewDir.getAbsolutePath());
+      Configuration previewhConf = new Configuration(hConf);
+      previewhConf.set(Constants.CFG_LOCAL_DATA_DIR, previewDir.getAbsolutePath());
+      previewcConf.setIfUnset(Constants.CFG_DATA_LEVELDB_DIR, previewDir.getPath());
+
+      return Guice.createInjector(
+        new ConfigModule(previewcConf, previewhConf),
+        new IOModule(),
+        new AuthenticationContextModules().getMasterModule(),
+        new SecurityModules().getStandaloneModules(),
+        new PreviewSecureStoreModule(secureStore),
+        new PreviewDiscoveryRuntimeModule(discoveryService),
+        new LocationRuntimeModule().getStandaloneModules(),
+        new ConfigStoreModule().getStandaloneModule(),
+        new PreviewModule(),
+        new ProgramRunnerRuntimeModule().getStandaloneModules(),
+        new PreviewDataModules().getDataFabricModule(transactionManager),
+        new PreviewDataModules().getDataSetsModule(datasetFramework, datasetNames),
+        new DataSetServiceModules().getStandaloneModules(),
+        new MetricsClientRuntimeModule().getStandaloneModules(),
+        new LoggingModules().getStandaloneModules(),
+        new NamespaceStoreModule().getStandaloneModules(),
+        new RemoteSystemOperationsServiceModule(),
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bind(ArtifactRepository.class).toInstance(artifactRepository);
+            bind(ArtifactStore.class).toInstance(artifactStore);
+            bind(AuthorizerInstantiator.class).toInstance(authorizerInstantiator);
+            bind(AuthorizationEnforcer.class).toInstance(authorizationEnforcer);
+            bind(PrivilegesManager.class).toInstance(privilegesManager);
+            bind(StreamAdmin.class).toInstance(streamAdmin);
+            bind(StreamConsumerFactory.class).to(InMemoryStreamConsumerFactory.class).in(Singleton.class);
+            bind(StreamCoordinatorClient.class).toInstance(streamCoordinatorClient);
+            bind(PreferencesStore.class).toInstance(preferencesStore);
+            // bind explore client to mock.
+            bind(ExploreClient.class).to(MockExploreClient.class);
+            bind(StorageProviderNamespaceAdmin.class).to(LocalStorageProviderNamespaceAdmin.class);
+          }
+
+          @Provides
+          @Named(Constants.Service.MASTER_SERVICES_BIND_ADDRESS)
+          @SuppressWarnings("unused")
+          public InetAddress providesHostname(CConfiguration cConf) {
+            String address = cConf.get(Constants.Service.MASTER_SERVICES_BIND_ADDRESS);
```

Comment:
Added `Constants.Preview.ADDRESS` and using it to bind here. In standalone this will be set to localhost. In Distributed, in `PreviewServerRunnable` it will be set using `TwillContext`.